test/logic/LocationTest.java

```diff
@@ -0,0 +1,240 @@
+        assertEquals(location.getName(), name);
+        assertEquals(location.isInitial(), isInitial);
+        assertEquals(location.isUrgent(), isUrgent);
+        assertEquals(location.isUniversal(), isUniversal);
+        assertEquals(location.isInconsistent(), isInconsistent);
+        assertEquals(location.getX(), x);
+        assertEquals(location.getY(), y);
+        assertEquals(location.getChildren(), children);
+    }
+
+    @Test
+    public void createInitialLocationDefaultsCoordinatesToZeroZero() {
+        // Arrange
+        String name = "initial location";
+        Guard invariant = new TrueGuard();
+        boolean isInitial = true;
+        boolean isUrgent = false;
+        boolean isUniversal = false;
+        boolean isInconsistent = false;
+        int x = 0;
+        int y = 0;
+        List<Location> children = new ArrayList<>();
+
+        // Act
+        Location initialLocation = Location.createInitialLocation(
+                name, invariant, isUrgent, isUniversal, isInconsistent
+        );
+
+        // Assert
+        assertEquals(initialLocation.getName(), name);
+        assertEquals(initialLocation.isInitial(), isInitial);
+        assertEquals(initialLocation.isUrgent(), isUrgent);
+        assertEquals(initialLocation.isUniversal(), isUniversal);
+        assertEquals(initialLocation.isInconsistent(), isInconsistent);
+        assertEquals(initialLocation.getX(), x);
+        assertEquals(initialLocation.getY(), y);
+        assertEquals(initialLocation.getChildren(), children);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void createCompositionLocationShouldThrowIllegalArgumentIfNoChildrenWereProvided() {
+        // Arrange
+        List<Location> children = new ArrayList<>();
+
+        // Act
+        Location.createComposition(children);
+    }
+
+    @Test
+    public void createRandomCompositionLocation() {
+        // Arrange
+        Random random = new Random();
+        List<Location> expectedChildren = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            Guard invariant = new TrueGuard();
+            String name = Integer.toString(random.nextInt());
+            boolean isInitial = random.nextBoolean();
+            boolean isUrgent = random.nextBoolean();
+            boolean isUniversal = random.nextBoolean();
+            boolean isInconsistent = random.nextBoolean();
+            int x = random.nextInt(1000);
+            int y = random.nextInt(1000);
+            Location location = Location.create(
+                    name, invariant, isInitial, isUrgent, isUniversal, isInconsistent, x, y
+            );
+            expectedChildren.add(location);
+        }
+
+        Guard expectedInvariant = new AndGuard(
+                expectedChildren.stream().map(Location::getInvariantGuard).collect(Collectors.toList())
+        );
+        String expectedName = expectedChildren.stream().map(Location::getName).collect(Collectors.joining());
```

Comment:
What if some of the random children have the same name?

Reply:
Then the joined named (Expected name of the composite) would have the same name multiple times. The test would still pass.